app/controllers/storage_manager_controller.rb

```diff
@@ -387,7 +385,9 @@ def set_record_vars(sm, mode = nil)
     # sm.agent_type = @edit[:new][:agent_typ] && @edit[:new][:agent_typ] != "" ? @edit[:new][:agent_typ] : "SMIS"
     sm.zone = Zone.find_by_name(@edit[:new][:zone])
 
-    sm.update_authentication({:default => {:userid => @edit[:new][:userid], :password => @edit[:new][:password]}}, :save => (mode != :validate))
+    sm.update_authentication({:default => {:userid   => @edit[:new][:userid],
+                                           :password => @edit[:new][:password] ? @edit[:new][:password] : sm.authentication_password}},
```

Comment:
So this sets the original value back in?  Why set it at all if nothing has changed, or do you not know that at this point?